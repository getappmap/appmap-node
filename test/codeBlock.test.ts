import { spawnSync } from "node:child_process";

import { integrationTest, readAppmaps, resolveTarget, runAppmapNode } from "./helpers";

integrationTest("mapping code block recording", () => {
  expect(runAppmapNode("index.js").status).toBe(0);
  expect(readAppmaps()).toMatchSnapshot();
});

integrationTest("throws when not run with appmap-node", () => {
  const testDir = resolveTarget();
  const result = spawnSync(process.argv[0], ["index.js"], { cwd: testDir });
  expect(result.stderr?.toString()).toContain(
    "Code is not instrumented. Please run the project with appmap-node.",
  );
});
