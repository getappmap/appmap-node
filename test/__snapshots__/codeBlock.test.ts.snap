// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mapping code block recording 1`] = `
{
  "./tmp/appmap/block/<timestamp 0>.appmap.json": {
    "classMap": [
      {
        "children": [
          {
            "children": [
              {
                "location": "index.js:5",
                "name": "hello",
                "static": true,
                "type": "function",
              },
            ],
            "name": "index",
            "type": "class",
          },
        ],
        "name": "codeBlock",
        "type": "package",
      },
    ],
    "events": [
      {
        "defined_class": "index",
        "event": "call",
        "id": 1,
        "lineno": 5,
        "method_id": "hello",
        "parameters": [
          {
            "class": "String",
            "name": "message",
            "value": "'world'",
          },
        ],
        "path": "index.js",
        "static": true,
        "thread_id": 0,
      },
      {
        "elapsed": 31.337,
        "event": "return",
        "id": 2,
        "parent_id": 1,
        "thread_id": 0,
      },
    ],
    "metadata": {
      "app": "codeBlock",
      "client": {
        "name": "appmap-node",
        "url": "https://github.com/getappmap/appmap-node",
        "version": "test node-appmap version",
      },
      "language": {
        "engine": "Node.js",
        "name": "javascript",
        "version": "test node version",
      },
      "name": "<timestamp 0>",
      "recorder": {
        "name": "block",
        "type": "block",
      },
    },
    "version": "1.12",
  },
  "./tmp/appmap/block/<timestamp 1>.appmap.json": {
    "classMap": [
      {
        "children": [
          {
            "children": [
              {
                "location": "index.js:5",
                "name": "hello",
                "static": true,
                "type": "function",
              },
            ],
            "name": "index",
            "type": "class",
          },
        ],
        "name": "codeBlock",
        "type": "package",
      },
    ],
    "events": [
      {
        "defined_class": "index",
        "event": "call",
        "id": 1,
        "lineno": 5,
        "method_id": "hello",
        "parameters": [
          {
            "class": "String",
            "name": "message",
            "value": "'world async'",
          },
        ],
        "path": "index.js",
        "static": true,
        "thread_id": 0,
      },
      {
        "elapsed": 31.337,
        "event": "return",
        "id": 2,
        "parent_id": 1,
        "thread_id": 0,
      },
    ],
    "metadata": {
      "app": "codeBlock",
      "client": {
        "name": "appmap-node",
        "url": "https://github.com/getappmap/appmap-node",
        "version": "test node-appmap version",
      },
      "language": {
        "engine": "Node.js",
        "name": "javascript",
        "version": "test node version",
      },
      "name": "<timestamp 1>",
      "recorder": {
        "name": "block",
        "type": "block",
      },
    },
    "version": "1.12",
  },
}
`;
